pipeline/src/main/java/edu/rosehulman/aixprize/pipeline/annotators/ColorsAnnotator.java

```diff
@@ -0,0 +1,98 @@
+package edu.rosehulman.aixprize.pipeline.annotators;
+
+import java.io.*;
+import java.net.*;
+
+import org.apache.uima.UimaContext;
+import org.apache.uima.analysis_component.JCasAnnotator_ImplBase;
+import org.apache.uima.analysis_engine.AnalysisEngineProcessException;
+import org.apache.uima.jcas.JCas;
+import org.apache.uima.jcas.tcas.Annotation;
+import org.apache.uima.json.JsonCasSerializer;
+import org.apache.uima.resource.ResourceInitializationException;
+import org.json.JSONObject;
+
+import edu.rosehulman.aixprize.pipeline.types.Color;
+
+/**
+ * Communicates with a Python script to find color words in a file.
+ * Interface:
+ * - writes a standard JSON-ified CAS to the Socket at port 8080
+ * - receives a series of annotations in JSON, one per line, with fields of:
+ *   + begin: int
+ *   + end: int
+ *   + name: String
+ * - the last annotation line should be an empty JSON object
+ */
+public class ColorsAnnotator extends JCasAnnotator_ImplBase {
+	
+	private Socket socket;
+
+	@Override
+	public void initialize(UimaContext aContext) throws ResourceInitializationException {
+		try {
+			socket = new Socket("127.0.0.1", 8080);
+		} catch (IOException e) {
+			e.printStackTrace();
+		}
+	}
+
+	@Override
+	public void process(JCas cas) throws AnalysisEngineProcessException {
+		if (socket == null) {
+			return;
+		}
+
+		try {
+			transmitCas(cas);
+			receiveAnnotations(cas);
+		} catch (IOException e) {
+			e.printStackTrace();
+		}
+	}
+
+	private void transmitCas(JCas cas) throws IOException {
+		StringWriter serialized = new StringWriter();
+		JsonCasSerializer.jsonSerialize(cas.getCas(), serialized);
+
+		byte[] casBytes = serialized.toString().getBytes();
```

Comment:
This and the next few lines are really gross. We need a way on the Python side to tell when the JSON has ended and we can stop reading from the socket. Using a higher-level protocol like HTTP would handle things like this for us - something to consider.